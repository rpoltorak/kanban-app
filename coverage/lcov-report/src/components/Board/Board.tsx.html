<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Board/Board.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Board</a> Board.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch, AnyAction } from "redux";
&nbsp;
import { CreateLaneAction, createLane } from "../../actions/laneActions";
import { RootState } from "../../reducers/rootReducer";
&nbsp;
import Lane from "../Lane";
import NewLane from "../NewLane";
&nbsp;
import "./Board.scss";
&nbsp;
interface BoardStateProps {
  lanes: string[];
}
&nbsp;
interface BoardDispatchProps {
  createLane: (title: string, description: string) =&gt; CreateLaneAction;
}
&nbsp;
type BoardProps = BoardStateProps &amp; BoardDispatchProps;
&nbsp;
export const Board: React.SFC&lt;BoardProps&gt; = (props: BoardProps) =&gt; (
  &lt;div className="board"&gt;
    &lt;div className="board__content"&gt;
      {props.lanes.map((laneId: string) =&gt; (
        &lt;Lane key={laneId} id={laneId} /&gt;
      ))}
      &lt;NewLane /&gt;
    &lt;/div&gt;
  &lt;/div&gt;
);
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(</span>state: RootState): BoardStateProps =&gt; <span class="cstat-no" title="statement not covered" >({</span>
  lanes: state.lanes.allIds,
});
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >(</span>dispatch: Dispatch&lt;AnyAction&gt;): BoardDispatchProps =&gt; <span class="cstat-no" title="statement not covered" >(</span>
  bindActionCreators({
    createLane,
  }, dispatch)
);
&nbsp;
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Board);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
