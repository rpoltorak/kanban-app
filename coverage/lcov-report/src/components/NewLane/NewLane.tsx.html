<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/NewLane/NewLane.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/NewLane</a> NewLane.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.49% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>32/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/24</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch, AnyAction } from "redux";
&nbsp;
import { createLane, CreateLaneAction } from "../../actions/laneActions";
import Editable from "../Editable";
&nbsp;
import "../Lane/Lane.scss";
&nbsp;
interface NewLaneDispatchProps {
  createLane: (title: string) =&gt; CreateLaneAction;
}
&nbsp;
interface NewLaneState {
  title: string;
  editMode: boolean;
}
&nbsp;
const initialState = {
  title: "",
  editMode: false,
};
&nbsp;
export class NewLane extends React.Component&lt;NewLaneDispatchProps, NewLaneState&gt; {
  public state = Object.assign({}, initialState);
&nbsp;
  public updateTitle = <span class="fstat-no" title="function not covered" >(</span>title: string): void =&gt; <span class="cstat-no" title="statement not covered" >(</span>
    this.setState(<span class="fstat-no" title="function not covered" >(</span>state: NewLaneState) =&gt; <span class="cstat-no" title="statement not covered" >({ title }))</span>
  )
&nbsp;
  public reset = () =&gt; this.setState((state: NewLaneState) =&gt; Object.assign({}, initialState));
&nbsp;
  public handleChange = <span class="fstat-no" title="function not covered" >(</span>title: string) =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >(</span>state: NewLaneState) =&gt; <span class="cstat-no" title="statement not covered" >({title}))</span>;</span>
  }
&nbsp;
  public handleCreate = () =&gt; {
    this.toggleEditMode();
    this.props.createLane(this.state.title);
  }
&nbsp;
  public toggleEditMode = () =&gt; this.setState((state: NewLaneState) =&gt; ({ editMode: !state.editMode}));
&nbsp;
  public renderCreateButton = () =&gt; (
    &lt;button className="btn btn--new-lane" onClick={this.toggleEditMode}&gt;+ Add new lane...&lt;/button&gt;
  )
&nbsp;
  public renderEditMode = () =&gt; (
    &lt;div className="lane"&gt;
      &lt;Editable
        placeholder="title"
        onChange={this.handleChange}
        maxLength={30}
        minLength={1}
        autoFocus
      &gt;&lt;/Editable&gt;
      &lt;button className="btn btn--save" onFocus={this.handleCreate}&gt;SAVE&lt;/button&gt;
      &lt;button className="btn btn--cancel" onClick={this.reset}&gt;CANCEL&lt;/button&gt;
    &lt;/div&gt;
  )
&nbsp;
  public render = () =&gt; this.state.editMode ? this.renderEditMode() : this.renderCreateButton();
}
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >(</span>dispatch: Dispatch&lt;AnyAction&gt;): NewLaneDispatchProps =&gt; <span class="cstat-no" title="statement not covered" >(</span>
  bindActionCreators({
    createLane,
  }, dispatch)
);
&nbsp;
export default connect(null, mapDispatchToProps)(NewLane);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 23:33:21 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
