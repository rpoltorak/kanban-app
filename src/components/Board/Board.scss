.board {
  display: flex;
  flex-direction: column;

  &__content {
    flex: 1;

    display: flex;
    flex-direction: row;
    align-items: flex-start;

    margin-bottom: 20px;
  }
}
