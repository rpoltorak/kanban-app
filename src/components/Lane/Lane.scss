.lane {
  position: relative;
  background-color: #dfe3e6;
  width: 220px;
  min-height: 30px;
  color: #000;
  border-radius: 0 0 3px 3px;
  padding: 15px;
  margin-right: 20px;

  &__title {
    .editable {
      font-size: 16px;
      font-weight: bold;
    }
  }
}
