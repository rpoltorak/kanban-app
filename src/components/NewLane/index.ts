import NewLane from "./NewLane";

export default NewLane;
