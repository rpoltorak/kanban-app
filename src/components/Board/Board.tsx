import React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch, AnyAction } from "redux";

import { CreateLaneAction, createLane } from "../../actions/laneActions";
import { RootState } from "../../reducers/rootReducer";

import Lane from "../Lane";
import NewLane from "../NewLane";

import "./Board.scss";

interface BoardStateProps {
  lanes: string[];
}

interface BoardDispatchProps {
  createLane: (title: string, description: string) => CreateLaneAction;
}

type BoardProps = BoardStateProps & BoardDispatchProps;

export const Board: React.SFC<BoardProps> = (props: BoardProps) => (
  <div className="board">
    <div className="board__content">
      {props.lanes.map((laneId: string) => (
        <Lane key={laneId} id={laneId} />
      ))}
      <NewLane />
    </div>
  </div>
);

const mapStateToProps = (state: RootState): BoardStateProps => ({
  lanes: state.lanes.allIds,
});

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>): BoardDispatchProps => (
  bindActionCreators({
    createLane,
  }, dispatch)
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Board);
