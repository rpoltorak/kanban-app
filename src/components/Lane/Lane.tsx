import React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch, AnyAction } from "redux";

import { DeleteLaneAction, deleteLane, updateLane, UpdateLaneAction } from "../../actions/laneActions";
import { RootState } from "../../reducers/rootReducer";
import { getLaneById } from "../../selectors/lanesSelector";

import NewCard from "../NewCard";
import Card from "../Card";
import Editable from "../Editable/Editable";

import "./Lane.scss";

interface LaneOwnProps {
  id: string;
}

interface LaneStateProps {
  title: string;
  cards: string[];
}

interface LaneDispatchProps {
  deleteLane: () => DeleteLaneAction;
  updateTitle: (title: string) => UpdateLaneAction;
}

type LaneProps = LaneOwnProps & LaneStateProps & LaneDispatchProps;

export const Lane: React.SFC<LaneProps> = (props: LaneProps) => (
  <div className="lane">
    <div className="lane__title">
      <Editable value={props.title} placeholder="title" onChange={props.updateTitle}></Editable>
      <button className="btn--delete" onClick={props.deleteLane}>&times;</button>
    </div>
    <div>
      {props.cards.map((cardId: string) => <Card key={cardId} id={cardId} laneId={props.id} />)}
    </div>
    <NewCard laneId={props.id} />
  </div>
);

const mapStateToProps = (state: RootState, ownProps: LaneOwnProps): LaneStateProps => {
  const { title, cards } = getLaneById(state, ownProps.id);

  return { title, cards };
};

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>, ownProps: LaneOwnProps): LaneDispatchProps => (
  bindActionCreators({
    deleteLane: () => deleteLane(ownProps.id),
    updateTitle: (title: string) => updateLane(ownProps.id, { title }),
  }, dispatch)
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Lane);
