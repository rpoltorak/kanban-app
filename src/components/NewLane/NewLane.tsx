import React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch, AnyAction } from "redux";

import { createLane, CreateLaneAction } from "../../actions/laneActions";
import Editable from "../Editable";

import "../Lane/Lane.scss";

interface NewLaneDispatchProps {
  createLane: (title: string) => CreateLaneAction;
}

interface NewLaneState {
  title: string;
  editMode: boolean;
}

const initialState = {
  title: "",
  editMode: false,
};

export class NewLane extends React.Component<NewLaneDispatchProps, NewLaneState> {
  public state = Object.assign({}, initialState);

  public updateTitle = (title: string): void => (
    this.setState((state: NewLaneState) => ({ title }))
  )

  public reset = () => this.setState((state: NewLaneState) => Object.assign({}, initialState));

  public handleChange = (title: string) => {
    this.setState((state: NewLaneState) => ({title}));
  }

  public handleCreate = () => {
    this.toggleEditMode();
    this.props.createLane(this.state.title);
  }

  public toggleEditMode = () => this.setState((state: NewLaneState) => ({ editMode: !state.editMode}));

  public renderCreateButton = () => (
    <button className="btn btn--new-lane" onClick={this.toggleEditMode}>+ Add new lane...</button>
  )

  public renderEditMode = () => (
    <div className="lane">
      <Editable
        placeholder="title"
        onChange={this.handleChange}
        maxLength={30}
        minLength={1}
        autoFocus
      ></Editable>
      <button className="btn btn--save" onFocus={this.handleCreate}>SAVE</button>
      <button className="btn btn--cancel" onClick={this.reset}>CANCEL</button>
    </div>
  )

  public render = () => this.state.editMode ? this.renderEditMode() : this.renderCreateButton();
}

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>): NewLaneDispatchProps => (
  bindActionCreators({
    createLane,
  }, dispatch)
);

export default connect(null, mapDispatchToProps)(NewLane);
