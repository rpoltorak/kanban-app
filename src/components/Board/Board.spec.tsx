import React from "react";
import { shallow } from "enzyme";

import { Board } from "./Board";
import NewLane from "../NewLane";
import Lane from "../Lane";

const lanes = ["1", "2", "3"];

describe("Board Component", () => {
  test("to match snapshot", () => {
    const board = shallow(
      <Board lanes={lanes} createLane={jest.fn()} />,
    );

    expect(board).toMatchSnapshot();
  });

  test("should render correctly when passed props", () => {
    const board = shallow(
      <Board lanes={lanes} createLane={jest.fn()} />,
    );

    expect(board.find(".board").exists()).toBe(true);
    expect(board.find(".board__content").exists()).toBe(true);
    expect(board.find(NewLane).exists()).toBe(true);
    expect(board.find(Lane).length).toBe(lanes.length);
  });
});
