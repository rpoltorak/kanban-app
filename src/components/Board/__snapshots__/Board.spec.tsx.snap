// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    createLane={[MockFunction]}
    lanes={
      Array [
        "1",
        "2",
        "3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="board__content"
      >
        <Connect(Component)
          id="1"
        />
        <Connect(Component)
          id="2"
        />
        <Connect(Component)
          id="3"
        />
        <Connect(NewLane) />
      </div>,
      "className": "board",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Connect(Component)
              id="1"
            />,
            <Connect(Component)
              id="2"
            />,
            <Connect(Component)
              id="3"
            />,
          ],
          <Connect(NewLane) />,
        ],
        "className": "board__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "id": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "id": "2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "id": "3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="board__content"
        >
          <Connect(Component)
            id="1"
          />
          <Connect(Component)
            id="2"
          />
          <Connect(Component)
            id="3"
          />
          <Connect(NewLane) />
        </div>,
        "className": "board",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <Connect(Component)
                id="1"
              />,
              <Connect(Component)
                id="2"
              />,
              <Connect(Component)
                id="3"
              />,
            ],
            <Connect(NewLane) />,
          ],
          "className": "board__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "id": "1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "id": "2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "id": "3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
